pos_fix_search_limit/static/src/js/pos_fix_search_limit.js

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2018 Akretion - Raphaël Reverdy
+   License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl). */
+odoo.define("pos_fix_search_limit.db", function (require) {
+    "use strict";
+    var PosDB = require("point_of_sale.DB");
+    PosDB.include({
+        limit: 314159265,  // the maximum number of results returned by a search
+    });     
+});
+
+odoo.define('pos_fix_search_limit.product_list', function(require) {
+    "use strict";
+    var screens = require('point_of_sale.screens');
+    var core = require('web.core');
+    var qweb = core.qweb;
+
+    screens.ProductListWidget.include({
+        displayLimit: 100, //number of elements displayed
+        renderElement: function() {
```

Comment:
Isn't there any better way than monkeypatching the method?

Reply:
@chienandalu Thank you for reviewing. I did not change any code as we use the v10 in production with odoo v12.
I guess the code can be refactored but I'm sorry not to have any time for that purpose.
Maybe Raphaël Reverdy <raphael.reverdy@akretion.com> who is the author of the module can answer

Reply:
> Isn't there any better way than monkeypatching the method?

I read the code and I don't see how it could be made better. We need to change the value of len in case of we need more elements than products (ie. if we use pos_product_template )
The module has been added here https://github.com/OCA/pos/pull/268

Reply:
Hi @chienandalu. Are you ok with the answer ? 

regards.

Reply:
Ok, I don't want to block it :+1: